pythx/models/response/base.py

```diff
@@ -1,8 +1,21 @@
 import abc
 import json
 
+import jsonschema
+
+from pythx.models.exceptions import ResponseValidationError
+
 
 class BaseResponse(abc.ABC):
+    schema = None
+
+    @classmethod
+    def validate(cls, candidate):
```

Comment:
So even when derived classes call `BaseResponse`'s implementation of `validate`, `cls.schema` would be the child's schema right?
ie: `VersionResponse` would have "version.json" as it's class schema.

Reply:
Exactly. `schema` is a class property, meaning that it's the same (pointer to) the data for each class. `validate` then just hooks into that data and interprets it as a schema.
Because the base classes are abstract, we cannot instantiate them directly, so as long as our concrete class definitions override `schema` with the stuff they need, we should be fine.

Reply:
However, there's a hidden bug here. Classes which don't override `schema` and don't implement `validate` will fail, because their schema variable is None and the default `validate` implementation is executed. So in the default implementation we should add another check for `None` and raise a validation error to notify the user about a missing schema

Reply:
Good catch